contracts/lib/SolidMath.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity 0.8.13;
+
+/**
+ * @dev Solid DAO Math Operations and Constants.
+ */
+abstract contract SolidMath {
+
+    uint256 constant public WEEKS_IN_SECONDS = 604800;
+
+    /**
+    @notice BASIS define in which basis value the interest rate per week must be informed
+    */
+    uint256 constant BASIS = 1000000;
+
+    /**
+     * @dev Calculates the number of weeks between a period
+     * @param _initialDate uint256 initial date to be informed in seconds
+     * @param _contractExpectedDueDate uint256 the carbon credit contract expected due date to be informed in seconds
+     */
+    function weeksInThePeriod(uint256 _initialDate, uint256 _contractExpectedDueDate) public pure returns(bool, uint256) {
+        if (_contractExpectedDueDate < WEEKS_IN_SECONDS) {
```

Comment:
I think this condition doesn't make sense. We need to check `_contractExpectedDueDate` in `depositReserveToken`. Something like:
```solidity
require(block.timestamp < carbonProjects[_token][_tokenId].contractExpectedDueDate, "Expected due date should be in the future.");
```

Reply:
It is this way because it was built to be used in other two functions we are going to build in the future, for selling and for rewarding investor. The decision to revert the transaction using require statement will belong to the function that calls this helper function.